src/server.js

```diff
@@ -23,8 +25,55 @@ const sendUserError = (err, res) => {
   }
 };
 
+const validateNameAndPass = (req, res, next) => {
+  const username = req.body.username;
+  const password = req.body.password;
+  if (!username || !password) {
+    res.status(STATUS_USER_ERROR);
+    res.json({ error: 'Must provide username and password' });
+    return;
+  }
+  req.username = username;
+  req.password = password;
+  next();
+};
+
 // TODO: implement routes
 
+// User signup
+server.post('/users', validateNameAndPass, (req, res) => {
+  bcrypt.hash(req.password, BCRYPT_COST, (err, hash) => {
+    if (err) {
+      sendUserError();
```

Comment:
Make sure you're actually passing the `err` into the `sendUserError()` method, otherwise it won't print a helpful error.